### ISSUE TITLE
Common projects 

### ISSUE THREAD

**Issue by Shobhit-hub** [csdu/placement-forum#5]
If we make some common projects like sorting visualiser , to do list etc . In interview how can we talk about them 
For example if the interviewer asks why did you make this project, what's different in your version ,
How can we answer these questions because these projects are all almost same and don't have any extra features

**Comment by ankushkhanna1998** [comment]
### Here's my take

- Share about your learnings from the project. Interviewers are interested in knowing about what comparisons you did for choosing the technologies you've used, what were the trade offs (sometimes having a greatly active community serves as a good reason too). What you learnt about team development, what about Software Engineering in general, ...
- About what best practices you've applied, and where did you get these things from. See, it's very important to know the good ways of doing things, otherwise there's not much of a point in doing something.
- Should have at least some value addition in your version, something to highlight, and something making sense in such a way. At least something right, big or small doesn't matter much, but something to talk about. You should spend some time thinking about such things for the project you're going to talk about in an interview.
- [_Now this one's really important to know_] What does this project lack:
    - In terms of features: What features are missing? Which are the ones which could have been implemented? Which are the ones which should be implemented in a future release?...
    - Why does it lack these things? What were the reasons? Now here you should have valid reasons: development bandwidth issues, tight deadlines, small team, learning curve, ...

And yes, a todo list can bring up a lot of discussion. It is one the most basic CRUD application. Discussions can be on the choice of database, the schema, the existing set of features, the future release features, on the UI/UX, etc. Even being a simple CRUD application, it has got a lot to it.

So, if you look closely, you have a lot to talk about anything, not just a project. And the best part for a project is, you've really spent time developing it. So take some time, and think what all you can highlight.

In short, most projects are common projects. But still there's a lot of room to discuss, a lot of software principles to cover, a lot of calculated discussions to take. If you haven't chosen them and/or followed some other known project, still you can get to know about these decisions by reading some articles, or getting to know about the intuitions the original creator had. And it's really okay.

Thanks for reaching out.
Hope that helps.

**Comment by jatin69** [comment]
+1 on the above points ankush brought up. To add to it,

- All of this needs to be prepared. All these questions, tech design choices needs to be thought through before the interview. On the fly thinking works for some, doesn't work for most.
- Maybe take 1 hour per project. Sit down, go through a generic list of questions like why this why that, and note it all down for your future reference. Then before the interview day, you can just revise your thoughts.
- Would also like to mention - Projects are almost never a deal breaker. You will almost never get rejected because you couldn't answer the questions well. But a good project and a good discussion will definitely add to your positive points. 
- An initial 5 minute discussion where you prove your development proficiency, helps a lot when asking for hints, and with overall interview flow.
- Basically, there's no harm to have good project prepped. 3 hours for 3 projects, another 2 for other resume items. Another 3 for general behavioural question. You're sorted for the entire interview season in just 1 day.
